AI/VCAI/VCAI.cpp

```diff
@@ -2459,6 +2459,25 @@ Goals::TSubgoal VCAI::questToGoal(const QuestInfo & q)
 		}
 		case CQuest::MISSION_HERO:
 		{
+			if(q.obj->tempOwner == playerID)
+			{
+				if(q.obj->ID == Obj::HERO)
+				{
+					auto heroToProtect = cb->getHero(q.obj->id);
+
+					return sptr(Goals::GatherArmy().sethero(heroToProtect));

```

Comment:
I think you should return invalid goal here and setup protect goal some different way, because you can not complete this quest.